data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/ProducerTopicConfig.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+/**
+ * An extension of the {@link TopicConfig} specifically for
+ * producers to Kafka topics.
+ */
+public interface ProducerTopicConfig extends TopicConfig {
```

Comment:
nit: ProducerTopicConfig and ConsumerTopicConfig gives impression that there are 2 different topics one for producing and one for consuming data. may TopicProducerConfig or TopicConfigProducer may be better, but not a blocker.

Reply:
@hshardeesi , That suggestion makes sense and I hadn't seen it from that perspective until you called it out. I went ahead and renamed the files.